dronegen/common.go

```diff
@@ -86,6 +87,81 @@ type buildType struct {
 	windowsUnsigned bool
 }
 
+// Description provides a human-facing description of the artifact, e.g.:
+//   Windows 64-bit (tsh client only)
+//   Linux ARMv7 (32-bit)
+//   MacOS Intel .pkg installer
+func (b *buildType) Description(packageType string, extraQualifications ...string) string {
+	var result string
+
+	var os string
+	var arch string
+	var darwinArch string
+	var bitness int
+	var qualifications []string
+
+	switch b.os {
+	case "linux":
+		os = "Linux"
+	case "darwin":
+		os = "MacOS"
+	case "windows":
+		os = "Windows"
+	default:
+		panic("unhandled OS")
```

Comment:
I don't know if we want to panic. Can we return a proper error and include the actual `b.os` value in the error message for troubleshooting?

Reply:
We could return an error, but that introduces a lot of error propagation up the call stack (unless at some point we still `panic()` or `log.Fatal()`):
* `(*buildType).Description()`
* `tagCreateReleaseAssetCommands()`
* `tagPipeline()`
* `tagPipelines()`
* `main()`

All for what is currently the sole fallible step in generating the YAML itself.

I agree with improving the error message.